test/address-test.js

```diff
@@ -14,4 +14,39 @@ describe('Address', function() {
     const expect = 'hs1qd42hrldu5yqee58se4uj6xctm7nk28r70e84vx';
     assert.strictEqual(addr.toString('main'), expect);
   });
+
+  it('should check standardness of address', () => {
+    const addr = new Address();
+
+    // Standard p2wpkh
+    addr.version = 0;
+    addr.hash = Buffer.alloc(20);
+    assert(!addr.isUnknown());
+
+    // Standard p2wsh
+    addr.version = 0;
+    addr.hash = Buffer.alloc(32);
+    assert(!addr.isUnknown());
+
+    // nullData, any valid length
+    for (let i = 2; i <= 40; i++) {
+      addr.version = 31;
+      addr.hash = Buffer.alloc(i);
+      assert(!addr.isUnknown());
+    }
```

Comment:
The way that you are constructing the nulldata output, you could technically create one with an invalid length and still have it pass here. I think that is fine since invalid outputs are considered unknown.

Reply:
True, there is no specified data length for witness program version 31, and so that is outside the scope of `isUnknown()`. Notice however that consensus limits data to `2 >= length >= 40` in `Address.isValid()` and that's why I chose those values for the test.